src/app/modules/system-process/pages/peer-locations-map/peer-locations-map.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, OnInit } from '@angular/core';
+import { take } from 'rxjs/operators';
+import { GeoCoordinate, GeoLocatorService } from '../../../core/services/geo-locator.service';
+declare let L: any;
+
+@Component({
+  selector: 'app-peer-locations-map',
+  templateUrl: './peer-locations-map.component.html'
+})
+export class PeerLocationsMapComponent implements OnInit {
+
+  constructor(
+    private geoLocationService: GeoLocatorService
+  ) { }
+
+  ngOnInit(): void {
+    const locationMap = L.map('peer-locations-map').setView([48, 32], 2.6);
+    const mapIcon: any = L.icon({
+      iconUrl: 'https://prysmaticlabs.com/assets/Prysm.svg',
+      iconSize: [30, 60]
+    });
+
+    this.geoLocationService.getPeerCoordinates().pipe(
+      take(1)
+    ).subscribe(geoCoordnates => {
+      if (geoCoordnates) {
+        const cityMarker: any = {};
+        const cityCounter: any = {};
+        geoCoordnates.forEach((coord: GeoCoordinate) => {
+          if (!cityCounter[coord.city]) {
+            cityCounter[coord.city] = 1;
+            cityMarker[coord.city] = L.marker([coord.lat, coord.lon], {
+              icon: mapIcon
+            });
+            cityMarker[coord.city].bindTooltip(coord.city).addTo(locationMap);
+          }
+          else {
+            cityCounter[coord.city]++;
+            cityMarker[coord.city].bindTooltip(coord.city + ' *' + cityCounter[coord.city]);
+          }
+        });
+      }
+    });
```

Comment:
Just a thought based on what @rauljordan has mentioned to me, but refactoring this into an observable without logic in the `.subscribe()` method, and using `ngOnDestroy` could help prevent memory leaks. FWIW, here is my attempt at this refactor (feel free to utilize all or none of it):
```
import { Component, OnInit, OnDestroy } from '@angular/core';
import { take, tap, takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';
import { GeoCoordinate, GeoLocatorService } from '../../../core/services/geo-locator.service';
declare let L: any;

@Component({
  selector: 'app-peer-locations-map',
  templateUrl: './peer-locations-map.component.html'
})
export class PeerLocationsMapComponent implements OnInit, OnDestroy {

  constructor(
    private geoLocationService: GeoLocatorService
  ) { }

  private destroyed$: Subject<void> = new Subject<void>();

  ngOnInit(): void {
    const locationMap = L.map('peer-locations-map').setView([48, 32], 2.6);
    const mapIcon: any = L.icon({
      iconUrl: 'https://prysmaticlabs.com/assets/Prysm.svg',
      iconSize: [30, 60]
    });

    this.geoLocationService.getPeerCoordinates().pipe(
      take(1),
      tap(geoCoordnates => {
        if (!geoCoordnates) return geoCoordnates;
        const cityMarker: any = {};
        const cityCounter: any = {};
        geoCoordnates.forEach((coord: GeoCoordinate) => {
          if (!cityCounter[coord.city]) {
            cityCounter[coord.city] = 1;
            cityMarker[coord.city] = L.marker([coord.lat, coord.lon], {
              icon: mapIcon
            });
            cityMarker[coord.city].bindTooltip(coord.city).addTo(locationMap);
          }
          else {
            cityCounter[coord.city]++;
            cityMarker[coord.city].bindTooltip(coord.city + ' *' + cityCounter[coord.city]);
          }
        });
      }),
      takeUntil(this.destroyed$)
    ).subscribe();

    L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
      attribution: '© <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
    }).addTo(locationMap);
  }

  ngOnDestroy(): void {
    this.destroyed$.next();
    this.destroyed$.complete();
  }
}
```

Reply:
@wackerow : You are right that taking care of memory leaks is important on subscriptions.
Thats why the  take(1) is there ... this means that the subscription is only valid for one request and then resolves, so shouldn't have a memory leak

Reply:
This is fine, but would definitely prefer never doing any logic within the `subscribe()` function. If we need side-effects, let's prefer tap().